### ISSUE TITLE
doc | ibmcloud: Inconsistency in default values for `cos_service_instance_name `

### ISSUE THREAD

**Issue by stevenhorsman** [confidential-containers/cloud-api-adaptor#21]
From https://github.com/confidential-containers/cloud-api-adaptor/pull/18#discussion_r875737240:
> In the next section the doc says:
> > cos_service_instance_name is the COS Service Instance Name, optional, default to value based on your cluster_name > tfvar if not set, ${cluster_name}-cos-service-instance.
>
> I think we should make the defaults of these line up, otherwise it will get confusing.

We now have inconsistency in the default values of the `cos_service_instance_name` in the two sections it is used:
- In creating the COS instance we have:
  > cos_service_instance_name in variables.tf has the default value 
- In building the pod VM image and uploading it we have:
  > cos_service_instance_name is the COS Service Instance Name, optional, default to value based on your cluster_name tfvar if not set, ${cluster_name}-cos-service-instance.
  
This is not a good user experience, so we have a few choices:
- Either get rid of the defaults and stop them being optional altogether
- Change both defaults to `cos-image-instance`. I don't think there is a requirement for the cos service instance name to be unique across users, so this should work
- Change both defaults to `<cluster_name>-cos-image-instance`. Unless there is a way to read the `cluster_name` value from the cluster terraform.tfvars, I don't think this is a good solution, as there isn't another reason for cluster name to be required in the cos instance creation, so it's letting a user not supply one value by making them supply another
- Any other solution someone smarter than me can think of?

**Comment by stevenhorsman** [comment]
@yoheiueda - you have more knowledge of terraform than me, so do you have an opinion on which option is best?

**Comment by yoheiueda** [comment]
@stevenhorsman I prefer changing the default value to `cos-image-instance`.

I found the description of the default value `<cluster_name>-cos-image-instance` at
https://github.com/confidential-containers/cloud-api-adaptor/blob/471c0baa1385d3fc19ebf1be73b6e6b46b7a6798/ibmcloud/README.md?plain=1#L202

But the terraform configuration files uses a `null` value for the variable default value.
https://github.com/confidential-containers/cloud-api-adaptor/blob/471c0baa1385d3fc19ebf1be73b6e6b46b7a6798/ibmcloud/terraform/podvm-build/variables.tf#L23

So, do we need to update the both files?

**Comment by stevenhorsman** [comment]
@yoheiueda - thanks for the input. That matches what I'd expect.

I think that we need to update to the doc as you said and add a new default into https://github.com/confidential-containers/cloud-api-adaptor/blob/471c0baa1385d3fc19ebf1be73b6e6b46b7a6798/ibmcloud/terraform/podvm-build/variables.tf#L23

and remove the logic in https://github.com/confidential-containers/cloud-api-adaptor/blob/471c0baa1385d3fc19ebf1be73b6e6b46b7a6798/ibmcloud/terraform/podvm-build/main.tf#L12 which currently does the defaulting.

BTW - I think Jonah is probably going to be looking at this issue soon.

**Comment by Jonah-F** [comment]
README requires change to manual running of ansible playbook as current command would not work because that playbook no longer exists. 

```
> $ ansible-playbook -i ./inventory -u root ./playbook.yml
```

needs to be

```
> $ ansible-playbook -i ./inventory -u root ./kube-playbook.yml && ansible-playbook -i ./inventory -u root ./kata-playbook.yml


```

**Comment by stevenhorsman** [comment]
issue accidentally closed by the wrong issue being labelled in the PR #37 commit, so re-opening.

**Comment by stevenhorsman** [comment]
Closed with #29 merging
